xlators/nfs/server/src/nfs.h

```diff
@@ -101,6 +101,11 @@ struct nfs_state {
     char *rpc_statd_pid_file;
     gf_boolean_t rdirplus;
     uint32_t event_threads;
+
+#ifdef HAVE_LIBTIRPC
+    pthread_mutex_t svc_mutex;
+    bool svc_running;
```

Comment:
Out of curiosity, could we have used an atomic and GF_ATOMIC_BASE_CMP_SWAP() instead?

Reply:
Yes, you are right. I'll update the patch.